### PR TITLE
global: support Jinja templating for job args

### LINKED ISSUE #36: Templating for job args
We want to allow specifying job args in a style similar to GitHub actions workflow templates, e.g.:

```json
{
  "from": "{{ job.last_run.created.isoformat() }}",
}
```

These are resolved/rendered before a Run is created and are evaluated as Python literals (falling back to `str` in case of errors).

### PR DESCRIPTION
**Pull request by slint**
* Closes #36.

### CODE REVIEW

File: invenio_jobs/models.py [hunk 1]
```diff
@@ -44,12 +46,17 @@ class Job(db.Model, Timestamp):
     default_args = db.Column(JSON, default=lambda: dict(), nullable=True)
     schedule = db.Column(JSON, nullable=True)
 
-    # TODO: See if we move this to an API class
     @property
     def last_run(self):
         """Last run of the job."""
         return self.runs.order_by(Run.created.desc()).first()
 
+    def last_run_with_status(self, status):
```
Comment (alejandromumo) [inline invenio_jobs/models.py:54]: it assumes `status` is something that can be added to the query. Is this OK? 

for github we have something similar but the status can be `None`:

```python
q = self.releases if status is None else self.releases.filter_by(status=status)
```

File: invenio_jobs/models.py [hunk 2]
```diff
@@ -63,6 +70,13 @@ def parsed_schedule(self):
         elif stype == "interval":
             return timedelta(**schedule)
 
+    def render_args(self, **ctx):
+        """Render job arguments."""
+        args = deepcopy(self.default_args)
+        ctx.setdefault("job", self)
+        walk_dict_values(args, lambda val: eval_tpl_str(val, ctx))
```
Comment (alejandromumo) [inline invenio_jobs/models.py:77]: The idea is to render the arguments into a jinja template I assume? 

is it ok for  this logic be so close to the model?

File: invenio_jobs/models.py (hunk 1 above)
Comment (slint) [reply invenio_jobs/models.py:54]: Yes, I should add some checks for this one. It's either that or making `last_run` a method, but I want something that can be easily used e.g. in a template.

File: invenio_jobs/models.py (hunk 2 above)
Comment (slint) [reply invenio_jobs/models.py]: In the end, I added a `Run.create` classmethod, that also takes care of some of the implicit logic we were doing in the service and scheduler (e.g. setting the default queue).

I think we can still get away without an API layer, so keeping things in the model is somewhat fine. I've also made the conscious decision that template arguments are evaluated before a Run is created, and the results are stored as is. That means that a Run is "pre-baked" with whatever args were computed at the time it's created.

**Pull request merged**